### PR TITLE
Sj/caip multichain api spec tests ci

### PR DESCRIPTION
**Pull request by shanejonas**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27317?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/run-api-specs-multichain.ts [hunk 1]
```diff
@@ -43,7 +45,7 @@ async function main() {
       disableGanache: true,
       title: 'api-specs coverage',
     },
-    async ({ driver }: { driver: Driver }) => {
+    async ({ driver, extensionId }: any) => {
```
Comment (shanejonas) [inline test/e2e/run-api-specs-multichain.ts:48]: adding `extensionId: string` to this type gave an error so put in `any` for now

File: test/e2e/run-api-specs-multichain.ts [hunk 2]
```diff
@@ -236,6 +230,13 @@ async function main() {
         testCoverageResultsCaip27,
       );
 
+      const htmlReporter = new HtmlReporter({
+        autoOpen: !process.env.CI,
+        destination: `${process.cwd()}/html-report-multichain`,
+      });
+
+      await htmlReporter.onEnd({} as IOptions, joinedResults);
+
```
Comment (shanejonas) [inline test/e2e/run-api-specs-multichain.ts:239]: merged the 2 multichain reports into 1 using the `HtmlReporter` itself

File: test/e2e/webdriver/chrome.js [hunk 3]
```diff
@@ -110,6 +110,7 @@ class ChromeDriver {
     return {
       driver,
       extensionUrl: `chrome-extension://${extensionId}`,
+      extensionId,
```
Comment (shanejonas) [inline test/e2e/webdriver/chrome.js:113]: the code already assumed this existed in `buildWebDriver`

image

**Review by adonesky1 — APPROVED** [review]
Left one comment otherwise LGTM!

File: .circleci/config.yml [hunk 4]
```diff
@@ -874,7 +877,7 @@ jobs:
           at: .
       - run:
           name: Build extension for testing
-          command: yarn build:test
+          command: CHAIN_PERMISSIONS=1 BARAD_DUR=1 yarn build:test
```
[the remark this answers is not in the document]
Comment (shanejonas) [reply .circleci/config.yml:880]: I kept getting a spinning loader when i didnt have it.

We have to remove these flags from the regular ci build before merging the feature branch anyways.

**Pull request merged**